backend/src/main/java/com/celuveat/restaurant/application/RestaurantQueryFacade.java

```diff
@@ -57,4 +64,40 @@ private List<RestaurantImageQueryResponse> relocateImagesByCelebId(
                 .ifPresent(imageQueryResponse -> Collections.swap(images, 0, images.indexOf(imageQueryResponse)));
         return images;
     }
+
+    @Transactional(readOnly = true)
+    public Page<RestaurantQueryResponse> findAll(
+            RestaurantSearchCond restaurantSearchCond,
+            LocationSearchCond locationSearchCond,
+            Pageable pageable,
+            CustomOptional<Long> memberId
+    ) {
+        Page<RestaurantQueryResponse> result = memberId.mapIfPresentOrElse(
+                id -> restaurantQueryService.findAllWithMemberId(
+                        restaurantSearchCond,
+                        locationSearchCond,
+                        pageable,
+                        id),
+                () -> restaurantQueryService.findAll(restaurantSearchCond, locationSearchCond, pageable)
+        );
```

Comment:
```suggestion
        Page<RestaurantQueryResponse> aa = memberId.map(id -> restaurantQueryService.findAllWithMemberId(
                        restaurantSearchCond,
                        locationSearchCond,
                        pageable,
                        id))
                .orElse(restaurantQueryService.findAll(restaurantSearchCond, locationSearchCond, pageable));
```
이렇게 `map`이랑 `orElse`를 섞어서 쓸 수 있더라고.
혹시 이거때문에 `CustomeOptional`을 만들었을까??

Reply:
이거 메서드는 orElseGet으로 했고 값 리턴하는거는 orElse했어.

```
Page<RestaurantQueryResponse> restaurantQueryResponse = memberId.map(
                id -> restaurantQueryService.findAllWithMemberId(
                        restaurantSearchCond,
                        locationSearchCond,
                        pageable,
                        id
                )).orElseGet(() -> restaurantQueryService.findAll(restaurantSearchCond, locationSearchCond, pageable));

        return Optional.ofNullable(restaurantSearchCond.celebId())
                .map(celebId -> relocateByCelebId(restaurantQueryResponse, celebId))
                .orElse(restaurantQueryResponse);
```